File: java/datastructure/datastructure/Stack/Stack.java
package datastructure.Stack;

import datastructure.linkedList.SinglyLinkedList;
import datastructure.linkedList.LinkedListNode;
public class Stack<T> {
    private SinglyLinkedList<T> stack; 
    
    public Stack() {
        this.stack = new SinglyLinkedList<T>(); 
    }

    public void push(T element) {
        if(this.stack.getHead() == null) {
            this.stack.setHead(new LinkedListNode<T>(element)); 
        } else {
            LinkedListNode<T> head_node = this.stack.getHead(), new_node = new LinkedListNode<T>(element);
            new_node.setNext(head_node); 
            this.stack.setHead(new_node); 
        }
    }

    public boolean isEmpty() {
        return this.stack.getHead() == null; 
    }

    public T pop() {
        if(this.stack.getHead() == null) {
            return null; // return if node is not found. 
        } else {
            LinkedListNode<T> pop_node = this.stack.getHead(); 
            this.stack.setHead(this.stack.getHead().getNext()); 
            return pop_node.getData(); // return data.  
        }
    }

}

File: python/datastructure/datastructure/Stack/Stack.py
# filename : Stack.py 
from .exception import StackUnderFlow 

class Stack: 
    def __init__(self): 
        self.stack = [] # stack.
    
    def push(self, element): 
        self.stack.insert(0, element) # add element in o
        return True 
    
    def isEmpty(self): 
        return self.stack == []

    def pop(self): 
        if self.isEmpty(): 
            raise StackUnderFlow() # if stack if empty throw StackUnderFlow Exception. 
        else: 
            return self.stack.pop(0) # remove top element. 

    def size(self): 
        return len(self.stack) # return size of stack. 
            

